packages/beacon-node/src/chain/bls/verifySets.ts

```diff
@@ -0,0 +1,56 @@
+import bls from "@chainsafe/bls";
+import {WorkRequestSet} from "./types.js";
+import {deserializeSet} from "./utils.js";
+
+const MIN_SET_COUNT_TO_BATCH = 2;
+
+/**
+ * Verify signatures sets with batch verification or regular core verify depending on the set count.
+ * Abstracted in a separate file to be consumed by the threaded pool and the main thread implementation.
+ */
+export function verifySets(sets: WorkRequestSet[]): boolean {
+  try {
+    const deserialized = sets.map(deserializeSet);
+    if (sets.length >= MIN_SET_COUNT_TO_BATCH) {
+      return bls.Signature.verifyMultipleSignatures(deserialized);
+    }
+
+    // .every on an empty array returns true
+    if (sets.length === 0) {
+      throw Error("Empty signature set");
+    }
+
+    // If too few signature sets verify them without batching
+    return deserialized.every(({message, publicKey, signature}) => signature.verify(publicKey, message));
+  } catch (_) {
+    // A signature could be malformed, in that case fromBytes throws error
+    // blst-ts `verifyMultipleSignatures` is also a fallible operation if mul_n_aggregate fails
+    // see https://github.com/ChainSafe/blst-ts/blob/b1ba6333f664b08e5c50b2b0d18c4f079203962b/src/lib.ts#L291
+    return false;
+  }
+}
+
+export async function asyncVerifySets(sets: WorkRequestSet[]): Promise<boolean> {
+  try {
+    const deserialized = sets.map(deserializeSet);
+    if (sets.length >= MIN_SET_COUNT_TO_BATCH) {
+      return await bls.asyncVerifyMultipleSignatures(deserialized);
```

Comment:
no need `await` here, also add a header comment for this function

Reply:
Good catch. Save a loop!!!  Although for some reason eslint is grumpy here.... hmmm.  Not sure that seems correct.  Its my understanding that the await unwraps the inner promise when returning to the ultimate consumer....

<img width="905" alt="Screenshot 2024-04-09 at 9 26 58 PM" src="https://github.com/ChainSafe/lodestar/assets/18608739/fe316bd7-57f9-422b-bd23-673a20ccc98d">